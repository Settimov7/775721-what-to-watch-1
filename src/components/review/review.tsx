import * as React from 'react';
import * as moment from 'moment';

import {User} from '../../types';

interface Props {
  comment: string;
  user: User;
  date: string;
  rating: number;
}

export const Review: React.FunctionComponent<Props> = (props) => {
  const {comment, user, date, rating} = props;
  const momentDate = moment(date);

  return (
    <div className="review">
      <blockquote className="review__quote">
        <p className="review__text">{comment}</p>

        <footer className="review__details">
          <cite className="review__author">{user.name}</cite>
          <time className="review__date" dateTime={momentDate.format(`YYYY-MM-DD`)}>{momentDate.format(`MMMM DD, YYYY`)}</time>
        </footer>
      </blockquote>

      <div className="review__rating">{rating}</div>
    </div>
  );
};
