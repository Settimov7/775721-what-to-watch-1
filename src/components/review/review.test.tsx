import * as React from 'react';
import * as renderer from 'react-test-renderer';

import {Review} from './review';

it(`Review correctly renders`, () => {
  const props = {
    user: {
      id: 1,
      name: `name`,
    },
    rating: 7,
    comment: `comment`,
    date: "2019-06-18T11:52:42.143Z"
  };

  const mainPage = renderer.create(
    <Review {...props}/>
  ).toJSON();

  expect(mainPage).toMatchSnapshot();
});
