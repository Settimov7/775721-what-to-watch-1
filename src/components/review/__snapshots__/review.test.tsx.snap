// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review correctly renders 1`] = `
<div
  className="review"
>
  <blockquote
    className="review__quote"
  >
    <p
      className="review__text"
    >
      comment
    </p>
    <footer
      className="review__details"
    >
      <cite
        className="review__author"
      >
        name
      </cite>
      <time
        className="review__date"
        dateTime="2019-06-18"
      >
        June 18, 2019
      </time>
    </footer>
  </blockquote>
  <div
    className="review__rating"
  >
    7
  </div>
</div>
`;
